lib/rig_inbound_gateway/api_proxy/handler/kafka.ex

```diff
@@ -39,20 +39,25 @@ defmodule RigInboundGateway.ApiProxy.Handler.Kafka do
 
   # ---
 
-  @spec kafka_handler(any()) ::
+  @spec kafka_handler(Cloudevents.kafka_body(), Cloudevents.kafka_headers()) ::
           :ok
           | {:error, %{:__exception__ => true, :__struct__ => atom(), optional(atom()) => any()},
              any()}
-  def kafka_handler(message) do
-    with {:ok, body} <- Jason.decode(message),
-         {:ok, rig_metadata} <- Map.fetch(body, "rig"),
+  def kafka_handler(message, headers) do
+    with {:ok, event} <- Cloudevents.from_kafka_message(message, headers),
+         {:ok, rig_metadata} <- Map.fetch(event.extensions, "rig"),
          {:ok, correlation_id} <- Map.fetch(rig_metadata, "correlation"),
          {:ok, deserialized_pid} <- Codec.deserialize(correlation_id) do
       Logger.debug(fn ->
         "HTTP response via Kafka to #{inspect(deserialized_pid)}: #{inspect(message)}"
       end)
 
-      send(deserialized_pid, {:response_received, message})
+      data =
+        if headers == [],
+          do: Cloudevents.Format.Encoder.JSON.encode(event),
+          else: Jason.encode!(event.data)
+
+      send(deserialized_pid, {:response_received, data})
```

Comment:
is this only required because we're not using `cloudevents` everywhere yet? perhaps a comment would help future-us..

Reply:
Exactly, should be gone once we migrate everything to use the cloudevents lib. Would do it in separate PRs though.

Reply:
Wait, what do you mean? Sorry, thought it's a different part of the code.